### PR TITLE
Restructure the documentation

### PR DESCRIPTION
**Pull request by lildude**
## Description

The README file has grown to massive proportions making it quite daunting to read and harder to maintain.

This PR restructures the documentation to move most of the content out of the README file and move it to individual files withing a `/docs` directory.

As part of this re-org I've also taken the opportunity to discourage the use of the macOS-supplied version of Ruby due to the problems it causes when installing charlock-holmes.

I've also expanded on the releasing docs for GitHub staff.

_[Rest of the template has been removed as it doesn't apply]_

### CODE REVIEW

**Review by Alhadis — COMMENTED** [review]
I have conflicting feelings here. On one hand, this strikes me as the sensible, natural thing to do from both a maintenance and reader perspective.

On the other… well, I've seen a lot of folks deep-link to specific sections of Linguist's readme (both in discussions, docs, and commit logs), which means this change will introduce bit-rot. 😕 Moreover, there are tricks people use to make huge, single-page files more accessible and easier to navigate:

1.	__Tables of contents linking to each heading.__  
	When it starts getting too lengthy, [break it into columns](https://github.com/vlang/v/blob/master/doc/docs.md#table-of-contents).
2.	__“Back to top” links__.  
	I personally find them bothersome, but I see these a lot, so it's clear many folks like 'em/
3.	__Using `<details>` or `<details open>` to implement collapsible sections__.  
	This might work well for the _Troubleshooting_ section, since impatient readers will likely be skimming each subsection looking for their relevant problem.

File: docs/overrides.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Overrides
+
+Linguist supports a number of different custom override strategies for language definitions and file paths.
```
[the remark this answers is not in the document]
Comment (lildude) [reply docs/overrides.md:3]: Nice idea. 👍

**Comment by lildude** [comment]
> On the other… well, I've seen a lot of folks deep-link to specific sections of Linguist's readme (both in discussions, docs, and commit logs), which means this change will introduce bit-rot. 😕

I did think about this but I thought his was a small price to pay to properly segregate "function" from "support", hence I've included the very small docs ToC at the top of the README. This could be extended.

> Moreover, there are tricks people use to make huge, single-page files more accessible and easier to navigate:

Indeed, but that introduces more complexity and doesn't help with the long-term maintenance or expansion of the docs.

**Comment by Alhadis** [comment]
> I did think about this but I thought his was a small price to pay to properly segregate "function" from "support"

Now that you mention it, I never noticed how sharply divided the readme's content was.

👍 then.

**Review by Alhadis — APPROVED** [review]

File: docs/overrides.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lildude) [reply docs/overrides.md:3]: How's that look 👇 ?  I thought it fit best down after the blurb about creating the file. I also reordered the individual sections to match the alphabetical order of the summary table.

[the remark this answers is not in the document]
Comment (Alhadis) [reply docs/overrides.md:3]: Brilliant.

**Review by jen801 — COMMENTED** [review]
#5446

File: docs/overrides.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jen801) [reply docs/overrides.md:3]: 😀

**Review by jen801 — COMMENTED** [review]
#5202

**Pull request merged**